corehq/apps/userreports/reports/filters/values.py

```diff
@@ -393,29 +422,75 @@ def show_all(self):
     def show_none(self):
         return self.value == self.SHOW_NONE
 
+    @property
+    def _ancestor_filter(self):
+        ancestor_expression = self.filter.get('ancestor_expression')
+        if (not (self.show_all and self.show_none) and
```

Comment:
This might be a bit more clear by making this:

```python
if self.show_all or self.show_none:
   return
if not ancestor_expression or len(self.value) != 1:
   return

# add param
```